src/lighteval/models/base_model.py

```diff
@@ -345,6 +352,97 @@ def greedy_until_with_logits(
+            requests, desc="Greedy Multi Turn generation", position=1, leave=False, disable=self.disable_tqdm
+        ):
+            # NOTE: we are assuming all items in a batch behave similarly (same
+            # stop_tokens and max_tokens genrated) which is not necessarily
+            # the case! Because of that we only use batch size of 1
+            stop_tokens = request.stop_sequence
+            max_generated_tokens = request.generation_size
+            context = request.context[0]
+            max_context_size_allowed = self.max_length - max_generated_tokens
+
+            tokenized = self.tokenizer(
+                context,
+                padding=True,
+                truncation=True,
+                return_tensors="pt",
+                max_length=max_context_size_allowed,
+                add_special_tokens=self.add_special_tokens,
+            ).to(self.device)
+
+            prepared_batch = Batch(
+                input_ids=tokenized["input_ids"],
+                input_lengths=[len(item == 1) for item in tokenized["attention_mask"]],
+                input_mask=tokenized["attention_mask"],
+                truncated=[0] * len(tokenized["input_ids"]),
+                padded=[0] * len(tokenized["input_ids"]),
+            )
+
+            cur_reponses = self._generate(
+                batch=prepared_batch,
+                max_new_tokens=max_generated_tokens,
+                stop_tokens=stop_tokens,
+                returns_logits=False,
+            )
+
+            model_answers = [cur_reponses[0].result]
+
+            for i, multi_turn_context in enumerate(request.context[1:]):
+                multi_turn_context = multi_turn_context.format(model_response=model_answers[0])
+
+                tokenized = self.tokenizer(
+                    multi_turn_context,
+                    padding=True,
+                    truncation=True,
+                    return_tensors="pt",
+                    max_length=max_context_size_allowed,
+                    add_special_tokens=self.add_special_tokens,
+                ).to(self.device)
+
+                prepared_batch = Batch(
+                    input_ids=tokenized["input_ids"],
+                    input_lengths=[len(item == 1) for item in tokenized["attention_mask"]],
+                    input_mask=tokenized["attention_mask"],
+                    truncated=[0] * len(tokenized["input_ids"]),
+                    padded=[0] * len(tokenized["input_ids"]),
+                )
+
+                cur_reponses = self._generate(
+                    batch=prepared_batch,
+                    max_new_tokens=max_generated_tokens,
+                    stop_tokens=stop_tokens,
+                    returns_logits=False,
+                )
+
+                model_answers.append(cur_reponses[0].result)
+
+            results.append(
+                GenerateMultiTurnReturn(
+                    result=model_answers,
+                    input_tokens=[],
```

Comment:
why not store the input and generated tokens?

Reply:
I forgot :)